server/integration/endpoint/src/main/java/org/infinispan/server/endpoint/subsystem/RestService.java

```diff
@@ -106,7 +113,33 @@ public synchronized void start(StartContext startContext) throws StartException
             builder.host(socketAddress.getAddress().getHostAddress());
             builder.port(socketAddress.getPort());
          }
-         restServer = NettyRestServer.createServer(builder.build(), cacheManagerInjector.getValue());
+
+         Authenticator authenticator;
+         switch (authMethod) {
+            case BASIC: {
+               SecurityRealm authenticationRealm = authenticationSecurityRealm.getOptionalValue();
+               SecurityDomain restSecurityDomain = new BasicRestSecurityDomain(authenticationRealm);
+               authenticator = new BasicAuthenticator(restSecurityDomain, EncryptableServiceHelper.isSecurityEnabled(this), authenticationRealm.getName());
+               break;
+            }
+            case CLIENT_CERT: {
+               if (!EncryptableServiceHelper.isSecurityEnabled(this)) {
+                  throw ROOT_LOGGER.cannotUseCertificateAuthenticationWithoutEncryption();
+               }
+               SecurityRealm authenticationRealm = authenticationSecurityRealm.getOptionalValue();
+               SecurityDomain restSecurityDomain = new ClientCertRestSecurityDomain(authenticationRealm);
+               authenticator = new ClientCertAuthenticator(restSecurityDomain);
+               break;
+            }
+            case NONE: {
+               authenticator = null;
+               break;
+            }
+            default:
+               throw ROOT_LOGGER.restAuthMethodUnsupported(authMethod.toString());
```

Comment:
Ah yes was wondering where those went to :)  So eventually this could be changed over, but in the interim we need it.